pkg/kubectl/cmd/set/set_rule.go

```diff
@@ -0,0 +1,524 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package set
+
+import (
+	"fmt"
+	"io"
+	"sort"
+	"strings"
+
+	"github.com/spf13/cobra"
+
+	"k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/types"
+	utilerrors "k8s.io/apimachinery/pkg/util/errors"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/kubernetes/pkg/apis/rbac"
```

Comment:
We shouldn't be adding new deps on internal apis

Reply:
I have changed the types, but I still need to use some functions in the internal apis "k8s.io/kubernetes/pkg/apis/rbac". @pwittrock @deads2k

Reply:
I think we should change the helper function to manipulate on external types in stead of internal types. And they should also be moved out of the `k8s.io/kubernetes` to somewhere else, (maybe `k8s.io/api`).

Reply:
endless duplication of complex helper methods per version is not something I'm eager for us to maintain